comp/core/workloadmeta/filter.go

```diff
@@ -36,38 +47,65 @@ type FilterParams struct {
 //
 // Only events of the given type will be delivered. Use EventTypeAll to collect
 // data from all the event types. EventTypeAll is the default.
-func NewFilter(filterParams *FilterParams) *Filter {
-	var kindSet map[Kind]struct{}
-	kinds := filterParams.Kinds
-	if len(kinds) > 0 {
-		kindSet = make(map[Kind]struct{})
-		for _, k := range kinds {
-			kindSet[k] = struct{}{}
-		}
+func NewFilterBuilder() *FilterBuilder {
```

Comment:
Should we add some unit tests that create some filters and check that the different Match functions work correctly? I know that we currently don't have these tests, but it might be a good opportunity to add them.

Reply:
Added unit tests

thanks for the suggestion 👍